expo/expo
# [expotool] improve generate-bare-app tooling

# Why



This PR addresses four things that I find annoying while using `generate-bare-app` 

1. the `addPackage` does not find package dependencies to symlink automatically
2. the newly created project does not stage and commit all of its changes on initialization 
3. it can only copy templates from `npm` via the `expo-cli` scripts 
4. running the command requires user input to enter the package name and bundle id (this comes from the expo-cli usage) 


# How



1. Updated the script template to look for dependencies and include these in the symlinks when adding a package via `package:add` script

2. Add and commit git changes on initialization3. 

3. provide a `localTemplate` flag which will let you copy over a project from the local `templates` directory in the expo repo, instead of generating from `expo-cli` 

4. automatically specify package and bundleIdentifier so you are not prompted by the expotool to enter these

# Test Plan



`et gba expo-dev-client --name test-1 --outDir ~/tmp`

- Ensure it builds without prompting
- git is initialized and there are no unstaged chagnes
- `yarn package:add expo-updates`  and ensure expo update dependencies are added to the project (e.g. expo-eas-client and expo-structured-headers`)

`et gba expo-dev-client --name test-1 --outDir ~/tmp --localTemplate`
 
- ensure the project builds w/ a copy of the local bare template


# Checklist
